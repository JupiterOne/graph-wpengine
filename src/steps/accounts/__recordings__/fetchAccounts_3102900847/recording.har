{
  "log": {
    "_recordingName": "fetchAccounts",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "4af80a67b8b7715f75fc69566d0c1fb9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.wpengineapi.com/v1/user"
        },
        "response": {
          "bodySize": 119,
          "content": {
            "mimeType": "application/json",
            "size": 119,
            "text": "{\"id\":\"3937869a-09a0-4198-a1c7-d70c451c24fc\",\"email\":\"stefan@creativice.com\",\"first_name\":\"Stefan\",\"last_name\":\"Virag\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "119"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:28:42 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "0e1d179c-e95f-4fd1-bb5c-f2f5a2ba2bc5"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_n0pFhKIAMF53g="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2ecea-4df20f310d0da6a825d1b252"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 a0549c0c4be19f37804ee07fca51b10d.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "PEofLMwLC-krC5A1NIls-yxf1KfnUzSnwId0PjhLanbTNzmp2q9-sg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:28:42.194Z",
        "time": 669,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 669
        }
      },
      {
        "_id": "b5a41916ddb8fd6f9a367486771ee186",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/accounts?limit=10&offset=0"
        },
        "response": {
          "bodySize": 112,
          "content": {
            "mimeType": "application/json",
            "size": 112,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\",\"name\":\"j1inc\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "112"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:28:43 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "3578bc27-3838-4460-a26c-9ec3b6bcffab"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_n00HBeoAMFU9A="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2eceb-0fd0c4e4414a5daa040a0110"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 f6773e0e5d339061619a1aeb9d029153.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "podix2UvqsAW69pR9ut6AkR8v35SP3Ya7fWL5NdbGHDivqdXh5w2gg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:28:42.913Z",
        "time": 1067,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1067
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
