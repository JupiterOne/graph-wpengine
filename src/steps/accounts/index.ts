import {
  createDirectRelationship,
  Entity,
  IntegrationStep,
  IntegrationStepExecutionContext,
  RelationshipClass,
} from '@jupiterone/integration-sdk-core';

import { IntegrationConfig } from '../../config';
import { createAPIClient } from '../../client';
import {
  Entities,
  Relationships,
  IntegrationSteps,
  USER_ENTITY_KEY,
} from '../constants';
import { createAccountEntity } from './converters';

export async function fetchAccounts({
  instance,
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config);

  const userEntity = (await jobState.getData(USER_ENTITY_KEY)) as Entity;

  await apiClient.iterateAccounts(async (account) => {
    const accountEntity = createAccountEntity(account);

    await jobState.addEntity(accountEntity);

    if (userEntity) {
      await jobState.addRelationship(
        createDirectRelationship({
          _class: RelationshipClass.HAS,
          from: userEntity,
          to: accountEntity,
        }),
      );
    }
  });
}

export const accountSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: IntegrationSteps.ACCOUNTS,
    name: 'Fetch Accounts',
    entities: [Entities.ACCOUNT],
    relationships: [Relationships.USER_HAS_ACCOUNT],
    dependsOn: [IntegrationSteps.USER],
    executionHandler: fetchAccounts,
  },
];
