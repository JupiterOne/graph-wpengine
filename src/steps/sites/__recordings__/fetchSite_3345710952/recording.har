{
  "log": {
    "_recordingName": "fetchSite",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "b5a41916ddb8fd6f9a367486771ee186",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/accounts?limit=10&offset=0"
        },
        "response": {
          "bodySize": 112,
          "content": {
            "mimeType": "application/json",
            "size": 112,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\",\"name\":\"j1inc\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "112"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Sun, 20 Jun 2021 05:45:03 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "f38102d7-0990-4160-a8c8-52d77b43e2ea"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "BNZa1GT_oAMFuRQ="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60ced5de-4a6338e3347eaae726b76f1b"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 7c2235d494e9a3b4e39b70d44b3f8810.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "T_d5Z-tmGGa1mLV9EzBMEDzy7WNDyb01fSREff29RsX3emltv7DJIg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-20T05:44:57.636Z",
        "time": 640,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 640
        }
      },
      {
        "_id": "97123c15bf93205301118bdc565d93ea",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 349,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/sites?limit=10&offset=0"
        },
        "response": {
          "bodySize": 354,
          "content": {
            "mimeType": "application/json",
            "size": 354,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"4d91f2dd-9743-421e-9bcf-20dcf0a195b2\",\"name\":\"jupiterone\",\"account\":{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\"},\"installs\":[{\"id\":\"30c47931-a91d-468a-a5dc-b9711f06c60b\",\"name\":\"jupiterone\",\"environment\":\"production\",\"cname\":\"jupiterone.wpengine.com\",\"php_version\":\"7.3\",\"is_multisite\":false}]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "354"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Sun, 20 Jun 2021 05:45:03 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "d01c3719-9310-4da0-9818-e55bb99a0fc0"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "BNZa8H9yIAMF1PA="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60ced5df-3bd589e82ff57eda29797079"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 63767b9f162b678f9022132db50015af.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "vg_9X4KinXtoapPfI5o1CS3bvPZBLuRFq-ymdDD57KkDyg_oLKxmDw=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-20T05:44:58.307Z",
        "time": 644,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 644
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
