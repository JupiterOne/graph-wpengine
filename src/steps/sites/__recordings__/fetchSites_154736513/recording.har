{
  "log": {
    "_recordingName": "fetchSites",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "b5a41916ddb8fd6f9a367486771ee186",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/accounts?limit=10&offset=0"
        },
        "response": {
          "bodySize": 112,
          "content": {
            "mimeType": "application/json",
            "size": 112,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\",\"name\":\"j1inc\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "112"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:36:15 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "6d3e0dcc-35e4-4acb-b217-d08c141210c9"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_o7UFZpIAMF5SQ="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2eeae-2e895a1c75421ba07b6737e8"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 f6773e0e5d339061619a1aeb9d029153.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "t_uyrZClLNnE1iP-6z5uITJveYJnU0C1XERQRk2Gp254uaNOZkxgVw=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:36:13.853Z",
        "time": 1274,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1274
        }
      },
      {
        "_id": "97123c15bf93205301118bdc565d93ea",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 349,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/sites?limit=10&offset=0"
        },
        "response": {
          "bodySize": 354,
          "content": {
            "mimeType": "application/json",
            "size": 354,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"4d91f2dd-9743-421e-9bcf-20dcf0a195b2\",\"name\":\"jupiterone\",\"account\":{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\"},\"installs\":[{\"id\":\"30c47931-a91d-468a-a5dc-b9711f06c60b\",\"name\":\"jupiterone\",\"environment\":\"production\",\"cname\":\"jupiterone.wpengine.com\",\"php_version\":\"7.3\",\"is_multisite\":false}]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "354"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:36:15 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "0114d2f0-2a75-4d52-a2b2-61200387f337"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_o7bHE0IAMFX-Q="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2eeaf-554dbcab6ca9ac255f765b91"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 b81cb31a4bdcef18545ac67bb3f11b5c.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "0SrdHTSvGAD83UQwJ0bBSfvpn0ku4TXFS0kWInFaqt2sOxJpuAW0Nw=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:36:15.173Z",
        "time": 650,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 650
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
