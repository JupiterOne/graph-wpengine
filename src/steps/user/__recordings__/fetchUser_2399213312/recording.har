{
  "log": {
    "_recordingName": "fetchUser",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "4af80a67b8b7715f75fc69566d0c1fb9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.wpengineapi.com/v1/user"
        },
        "response": {
          "bodySize": 119,
          "content": {
            "mimeType": "application/json",
            "size": 119,
            "text": "{\"id\":\"3937869a-09a0-4198-a1c7-d70c451c24fc\",\"email\":\"stefan@creativice.com\",\"first_name\":\"Stefan\",\"last_name\":\"Virag\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "119"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Fri, 18 Jun 2021 15:07:33 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "d204b93a-2d7f-461f-bd8d-671e34cf15a3"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "BIF8THXhIAMFclw="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60ccb6b4-0596b255286e49bb1a1f1cef"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 881b12332738e10f6e80298fbdcd7e8f.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "VIE50-C2"
            },
            {
              "name": "x-amz-cf-id",
              "value": "MIjnbkgnc_Dl5H_XKCCMwLtognKlLK9VgZ0cdTWTCtwVXWu_iUrQ6Q=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-18T15:07:32.577Z",
        "time": 773,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 773
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
