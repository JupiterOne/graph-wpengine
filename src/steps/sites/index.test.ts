import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchSites } from '.';
import { integrationConfig } from '../../../test/config';
import { setupWPEngineRecording } from '../../../test/recording';
import { fetchAccounts } from '../accounts';

describe('#fetchSites', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupWPEngineRecording({
      directory: __dirname,
      name: 'fetchSites',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAccounts(context);
    await fetchSites(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const sites = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('Host'),
    );
    expect(sites.length).toBeGreaterThan(0);
    expect(sites).toMatchGraphObjectSchema({
      _class: ['Host'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'wp_engine_site' },
          name: { type: 'string' },
        },
      },
    });

    const accounts = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('Account'),
    );
    expect(accounts.length).toBeGreaterThan(0);
    expect(accounts).toMatchGraphObjectSchema({
      _class: ['Account'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'wp_engine_account' },
          name: { type: 'string' },
        },
      },
    });
  });
});
