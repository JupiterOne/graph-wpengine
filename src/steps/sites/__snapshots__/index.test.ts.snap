// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchSite should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "wp_engine_account:77815c5d-e25c-41a3-ba2d-29f8ceb82ac1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "id": "77815c5d-e25c-41a3-ba2d-29f8ceb82ac1",
            "name": "j1inc",
          },
        },
      ],
      "_type": "wp_engine_account",
      "createdOn": undefined,
      "displayName": "j1inc",
      "id": "77815c5d-e25c-41a3-ba2d-29f8ceb82ac1",
      "name": "j1inc",
    },
    Object {
      "_class": Array [
        "Host",
      ],
      "_key": "wp_engine_site:4d91f2dd-9743-421e-9bcf-20dcf0a195b2",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "account": Object {
              "id": "77815c5d-e25c-41a3-ba2d-29f8ceb82ac1",
            },
            "id": "4d91f2dd-9743-421e-9bcf-20dcf0a195b2",
            "installs": Array [
              Object {
                "cname": "jupiterone.wpengine.com",
                "environment": "production",
                "id": "30c47931-a91d-468a-a5dc-b9711f06c60b",
                "is_multisite": false,
                "name": "jupiterone",
                "php_version": "7.3",
              },
            ],
            "name": "jupiterone",
          },
        },
      ],
      "_type": "wp_engine_site",
      "createdOn": undefined,
      "displayName": "jupiterone",
      "id": "4d91f2dd-9743-421e-9bcf-20dcf0a195b2",
      "name": "jupiterone",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "wp_engine_account:77815c5d-e25c-41a3-ba2d-29f8ceb82ac1",
      "_key": "wp_engine_account:77815c5d-e25c-41a3-ba2d-29f8ceb82ac1|has|wp_engine_site:4d91f2dd-9743-421e-9bcf-20dcf0a195b2",
      "_toEntityKey": "wp_engine_site:4d91f2dd-9743-421e-9bcf-20dcf0a195b2",
      "_type": "wp_engine_account_has_site",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "wp_engine_account",
    "wp_engine_site",
    "wp_engine_account_has_site",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
